pyccel/codegen/printing/fcode.py

```diff
@@ -1301,7 +1301,8 @@ def _print_Assign(self, expr):
 
             # in the case of a function that returns a list,
             # we should append them to the procedure arguments
-            if isinstance(expr.lhs, (tuple, list, PythonTuple)):
+            if isinstance(expr.lhs, (tuple, list, PythonTuple, InhomogeneousTupleVariable)) \
+                    or (isinstance(expr.lhs, HomogeneousTupleVariable) and expr.lhs.is_stack_array):
```

Comment:
Do we have a test for the last case that you consider here?
```python
isinstance(expr.lhs, HomogeneousTupleVariable) and expr.lhs.is_stack_array
```

Reply:
I think [this](https://github.com/pyccel/pyccel/blob/master/tests/epyccel/modules/tuples.py#L238) should test it

Reply:
So it is tested whenever the output is multiple values of the same type, unless that type is str or similar that fortran can't handle in a tuple

Reply:
Tested [here](https://github.com/pyccel/pyccel/blob/master/tests/pyccel/scripts/runtest_multiple_results.py#L35)